qcog_python_client/qcog/pytorch/validate/_validate_model_module.py

```diff
@@ -111,4 +100,8 @@ def validate_model_module(
             arg_name=ann, arg_type=train_fn.__annotations__[ann]
         )
 
-    return ValidateModelModule(train_fn=train_fn_annotations)
+    # Set the train function annotations on the handler
+    setattr(self, "train_fn_annotations", train_fn_annotations)
+
+    # Directory has been validated
+    return copy.deepcopy(directory)
```

Comment:
If our validation may change the object don't we want to then have the object deepcopied at the beginning of the operation, because if it does modify the object then the way I'm imagining it is:

I pass you Directory
You modify Directory -> Directory', so now my pointed to Directory goes to Directory'
You then deepcopy Directory' and I now have two Directory' and no Directory

?

Reply:
oh right right, in this case is ok anyway cause I'm overwriting the reference to the original directory so I'm not creating any external side effect anyway, but you are correct